internal/lock.go

```diff
@@ -50,160 +49,199 @@ func NewLock(path string, newOk bool) (*Lock, error) {
 }
 
 func (l *Lock) AddResource(paths []string, algo string, tags []string, filename string) error {
+	return l.AddResourceWithCache(paths, algo, tags, filename, "")
+}
+
+func (l *Lock) AddResourceWithCache(paths []string, algo string, tags []string, filename string, cacheUri string) error {
```

Comment:
Instead of adding new methods like this, you can simply have one method and use no cache if cacheUri is the empty string. This will be simpler and keep the "interface" surface small.